lms/djangoapps/course_api/api.py

```diff
@@ -266,3 +267,90 @@ def get_course_run_url(request, course_id):
     """
     course_run_url = reverse('openedx.course_experience.course_home', args=[course_id])
     return request.build_absolute_uri(course_run_url)
+
+
+def get_course_members(course_key):
+    """
+    Returns a dict containing all users with access to a course.
+
+    User information includes:
+        - User info: id, email, username, name
+        - Enrollment mode
+        - Role list
+
+    Examples:
+        - Get all course members:
+            get_course_members(course_key)
+
+    Arguments:
+        course_key (CourseKey): CourseKey to retrieve student data.
+
+    Returns:
+        dict: A dictionary with the following format:
+            {
+                "user_id": {
+                    "id": 12,
+                    "username": "jonh5000",
+                    "email": "jonh@example.com",
+                    "name": "Jonh Doe",
+                    "enrollment_mode": "verified",
+                    "roles": [
+                        "student",
+                        "instructor",
+                        "staff",
+                    ]
+                }
+            }
+    """
+    def get_user_info(user, enrollment_mode=None):
+        """
+        Utility function to extract user information from model.
+        """
+        return {
+            "id": user.id,
+            "username": user.username,
+            "email": user.email,
+            "name": user.profile.name,
+            "enrollment_mode": enrollment_mode,
+        }
+
+    # Raise error if trying to retrieve user list from a course with more than
+    # settings.COURSE_MEMBER_API_ENROLLMENT_LIMIT active enrollments. The fastest way
+    # to do this is to query for the 1st item after `COURSE_MEMBER_API_ENROLLMENT_LIMIT`.
+    over_limit = CourseEnrollment.get_active_enrollments_in_course(
+        course_key
+    )[settings.COURSE_MEMBER_API_ENROLLMENT_LIMIT:][:1]
+    if over_limit.exists():
+        raise Exception(
+            f"Can't retrieve course members for courses with more than"
+            f"{settings.COURSE_MEMBER_API_ENROLLMENT_LIMIT} active enrollments"
+        )
```

Comment:
Should we add the course key to this error message?  Or will it be obvious from some other source in the logs?

Reply:
@nedbat Yes, good idea! The log output now look like this:
```
Can't retrieve course members for COURSE_KEY since it has more than 1000 active enrollments. This limit is stored on `settings.COURSE_MEMBER_API_ENROLLMENT_LIMIT`
```